### PR TITLE
api: adjustments to not require config as an arg

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

make sure the python script experience of using our api doesn't requiring touching any ua-internals like the UAConfig

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by orndorffgrant** [comment]
If we agree on this general approach, then I'll fix the various test failures

File: uaclient/api/u/pro/attach/magic/initiate/v1.py [hunk 1]
```diff
@@ -21,25 +22,29 @@ class MagicAttachInitiateResult(DataObject, AdditionalInfo):
...
 
 
-def initiate(cfg: UAConfig) -> MagicAttachInitiateResult:
+def initiate() -> MagicAttachInitiateResult:
+    return _initiate()
+
+
+def _initiate(*, cfg: Optional[UAConfig] = None) -> MagicAttachInitiateResult:
+    if cfg is None:
+        cfg = UAConfig()
```
Comment (lucasmoura) [inline uaclient/api/u/pro/attach/magic/initiate/v1.py:42]: I like this approach that covers both sides of using the API. I have no problems going forward with this refactoring

**Review by renanrodrigo — APPROVED** [review]
Nice!

**Review by CalvoM — APPROVED** [review]
LGTM, check the failing integration test.

**Comment by orndorffgrant** [comment]
Yep I made a silly mistake while editing that test. Should be fixed now but let's wait for CI to go green

**Pull request merged**